### PR TITLE
grpcproxy: fix more issues in watch path

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony 

Several more tests can pass now.

### CODE REVIEW

File: proxy/grpcproxy/watcher.go [hunk 1]
```diff
@@ -42,12 +42,13 @@ func (w *watcher) send(wr clientv3.WatchResponse) {
 
 	events := make([]*mvccpb.Event, 0, len(wr.Events))
 
+	var lastRev int64
 	for i := range wr.Events {
 		ev := (*mvccpb.Event)(wr.Events[i])
 		if ev.Kv.ModRevision <= w.rev {
 			continue
 		} else {
-			w.rev = ev.Kv.ModRevision
```
Comment (heyitsanthony) [inline proxy/grpcproxy/watcher.go]: why can't `w.rev` be directly updated here?

Comment (xiang90) [inline proxy/grpcproxy/watcher.go]: txn can have multiple events with the same rev. if we directly update here, we would skip the following events in the same txn.

Comment (heyitsanthony) [inline proxy/grpcproxy/watcher.go]: OK, add a comment?

**Comment by heyitsanthony** [comment]
lgtm

File: proxy/grpcproxy/watcher.go (hunk 1 above)
Comment (xiang90) [inline proxy/grpcproxy/watcher.go]: will do.

**Pull request merged**